### DISCUSSION TITLE
Unable to set up php support

### DISCUSSION THREAD

**Discussion by navxio** [LazyVim/LazyVim#1067] [Q&A] [answered]
I'm trying to set up neovim for php development.. thsi is my php.lua
```
return {
  {
    "williamboman/mason.nvim",
    opts = function(_, opts)
      vim.list_extend(opts.ensure_installed, {
        "php-cs-fixer",
        "phpactor",
      })
    end,
  },

  -- Add `server` and setup lspconfig
  {
    "neovim/nvim-lspconfig",
    dependencies = {},
    opts = {
      servers = {
        phpactor = {},
      },
    },
  },

  -- Setup null-ls with `phpcsfixer`
  {
    "jose-elias-alvarez/null-ls.nvim",
    opts = function(_, opts)
      local nls = require("null-ls")
      opts.sources = vim.list_extend(opts.sources, {
        nls.builtins.formatting.phpcsfixer,
      })
    end,
  },
}
```
Somehow mason installed intelephense and not phpactor.. what gives?

**Comment by jasalt** [comment, marked as answer]
Maybe you have some config reinstalling intelephense?

I was able to get intelephense uninstalled and phpactor installed from Mason (required installing composer from OS package manager) and phpactor running solely when editing php file when checking lsp info with `<leader>cl`. It just doesn't seem to work as far as intelephense, I'm getting error from go to definition command `gD`: `method textDocument/declaration is not supported by any of the servers registered for the current buffer` while intelephense works as expected.

Going to continue with intelephense for now while trying to figure how to configure it with `braces = "k&r"`...

There's also this other PHP discussion here mainly on the debugger setup https://github.com/LazyVim/LazyVim/discussions/645.
